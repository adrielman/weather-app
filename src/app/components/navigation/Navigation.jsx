import React from "react";
import { NavLink } from "react-router-dom";
import styles from "./Navigation.module.scss";

export const Navigation = ({ children }) => <nav className={styles.navigation}>{children}</nav>;

export const Link = ({ to, exact, children }) => (
  <div className={styles.link}>
    <NavLink to={to} exact={exact} activeClassName={styles.selected}>
      {children}
    </NavLink>
  </div>
);
