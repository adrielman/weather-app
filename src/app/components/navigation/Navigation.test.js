import React from "react";
import { shallow } from "enzyme";
import { Navigation, Link } from "./Navigation";

describe("Navigation", () => {
  const navigation = shallow(
    <Navigation>
      <Link to="/" exact>
        Home
      </Link>
      <Link to="/link1">Link1</Link>
      <Link to="/link2">Link2</Link>
    </Navigation>
  );

  it("should match snapshot", () => {
    expect(navigation).toMatchSnapshot();
  });

  it("link should match snapshot", () => {
    const link = shallow(<Link to="/link1">Link1</Link>);
    expect(link).toMatchSnapshot();
  });
});
