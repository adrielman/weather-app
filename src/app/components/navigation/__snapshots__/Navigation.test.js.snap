// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation link should match snapshot 1`] = `
<div
  className="link"
>
  <NavLink
    activeClassName="selected"
    aria-current="page"
    to="/link1"
  >
    Link1
  </NavLink>
</div>
`;

exports[`Navigation should match snapshot 1`] = `
<nav
  className="navigation"
>
  <Link
    exact={true}
    to="/"
  >
    Home
  </Link>
  <Link
    to="/link1"
  >
    Link1
  </Link>
  <Link
    to="/link2"
  >
    Link2
  </Link>
</nav>
`;
